docs/concepts.rst

```diff
@@ -271,3 +271,91 @@ Modules not yet Supported for Criteria
 :mod:`RunHistory <cbc_sdk.audit_remediation.base.RunHistory>`
 :mod:`Event <cbc_sdk.enterprise_edr.base.Event>`
 :mod:`Process <cbc_sdk.enterprise_edr.base.Process>`
+
+
+Asynchronous queries
+--------------------
+
+A number of queries allow for asynchronous mode of operation. Those utilize python futures and the request itself is performed in a separate worker thread.
+An internal thread pool is utilized to support multiple CBC queries executing in an asynchronous manner without blocking the main thread.
+
+Execute an asynchronous query
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Running asynchronous queries is done by invoking the ``execute_async()`` method, e.g:
+
+  >>> async_query = api.select(EnrichedEvent).where('process_name:chrome.exe').execute_async()
+
+The ``execute_async()`` method returns a python future object that can be later on waited for results.
+
+Fetching asynchronous queries' results
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Results from asynchronous queries can be retrieved by using the result() method since they are actually futures:
+
+  >>> print(async_query.result())
+
+This would block the main thread until the query completes.
+
+Modules with support for asynchronous queries
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+:mod:`Process <cbc_sdk.enterprise_edr.base.Process>`
```

Comment:
Could we add Process Facets to this list as well?

`:mod:ProcessFacet <cbc_sdk.enterprise_edr.base.ProcessFacet>`

Reply:
Added that as well